examples/extractor/example_extractor.yml

```
@@ -16,10 +16,28 @@ source_repository: >-
     https://github.com/example/extractor
 documentation: >-
     https://example.github.io/extractor
-usage: >-
-    - cli:parse --type=example {{ file_path }}
-    - python:example.extrators.extract({{ file_path }}, type="example")
-    - python:example.extrators.extract.ExampleExtractor({{ file_path }}).parse()
+usage:
+    - method: cli
+      command: parse --type=example {{ file_path }}
+    - method: python
+      setup: example.extractors
+      command: extractors.extract({{ file_path }}, type="example")
```

```suggestion
      command: example.extractors.extract({{ file_path }}, type="example")
```

I believe?

I guess we could also allow syntax like:

- `example.extractors as example`
- `ExampleExtractor from example.extractors`
- `* from example.extractors`

---

I think the original is correct (without `example.` given the `setup:` field above).

As for the second point, I think we'll be using `importilb.import_module()`, which means we're restricted to importing modules/packages. We will still need to figure out how to excise the function name out of the `command:`; it's perhaps better to not complicate things and allow only modules into `setup:`.

---

This is what I'm getting at for the first, replace `setup: scipy.signal` and command `signal.savgol_filter` and you get this:

```
>>> import scipy.signal
>>> signal.savgol_filter
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'signal' is not defined
>>> scipy.signal.savgol_filter
<function savgol_filter at 0x7f9bafc1d0d0>
```

or should I be reading it as `setup: scipy.signal` and `command: savgol_filter` needing to then be composed to `scipy.signal.savgol_filter`? In that case I think its just the plural that's incorrect:

```suggestion
      command: extractor.extract({{ file_path }}, type="example")
```

---

Fine for the second point though!

---

You're right. I'll commit the first suggestion.